### PR TITLE
fix(callout): allow Callout component to not have an icon

### PR DESCRIPTION
**Pull request by liaprins**
Callout component can now exist sans icon by adding icon={false} as a prop

re #577

## Summary

**Callout**
Github issue: #577 

Recommended fix (per @tihuan ), documenting here for posterity:
Change the [source code](https://github.com/chanzuckerberg/sci-components/blob/bdb41f9a1e4581e45d9adeda7633593585f717e0/packages/components/src/core/Callout/index.tsx#L59) from

`if (icon) return icon;`
to:
`if (icon !== undefined) return icon;`
"And then the following will work:
icon={false} - it will not take up space in the component. Similar to display: none;
icon={<></>} - it will still take up space in the component, but just empty space. Similar to visibility: hidden;
icon={null} - this is a little counter intuitive, but it will fall back to using MUI's iconMapping"

## Checklist

- [x] Default Story in Storybook
- [x] LivePreview Story in Storybook
- [x] Test Story in Storybook
- [x] Tests written
- [x] Variables from `defaultTheme.ts` used wherever possible
- [x] If updating an existing component, depreciate flag has been used where necessary
- [x] Chromatic build verified by @chanzuckerberg/sds-design

**Note that I am also having David Ruiz  help with reviewing bug fixes for SDS Hackday (which fixing this bug was part of), but I couldn't find him listed yet when I went to add reviewers**

### CODE REVIEW

**Comment by DRuizCZI** [comment]
LGTM but i will feel better after @timmy or @masoudmanson  Take a look

File: packages/components/src/core/Callout/index.tsx
Comment (tihuan) [inline packages/components/src/core/Callout/index.tsx:59]: Niice! Thanks so much for the fix, @liaprins 🙏 ❤️ 

Can we add a comment to explain why we explicitly use `icon !== undefined` here please? In case someone in the future "refactors" this back to `if (icon) return icon` lol

Something like:

```ts
/**
 * (liaprins): Explicitly use `icon !== undefined` to fix bug in ticket: https://github.com/chanzuckerberg/sci-components/issues/577 
 */
```

Or something better!

**Review by masoudmanson — APPROVED** [review]
Great catch, @liaprins! 
Thank you for addressing this issue. The change LGTM. 

However, before merging, let's ensure we add the comment mentioned by @tihuan. Once that's added, we're all set to merge this. Appreciate your quick action on this! 🎉⭐️

File: packages/components/src/core/Callout/index.tsx
Comment (liaprins) [reply packages/components/src/core/Callout/index.tsx]: Great, thanks for the suggestion! I have finally updated to add the comment as suggested!

**Pull request merged**
